src/platform_impl/windows/event_loop.rs

```diff
@@ -1939,7 +1939,7 @@ unsafe extern "system" fn thread_event_target_callback<T: 'static>(
     // the closure to catch_unwind directly so that the match body indendation wouldn't change and
     // the git blame and history would be preserved.
     let callback = || match msg {
-        winuser::WM_DESTROY => {
+        winuser::WM_NCDESTROY => {
```

Comment:
The documentation says `WM_DESTROY` is for destruction of the main window, while `WM_NCDESTROY` is for child windows.

Looking at the docs, it seems like winit should be using `WM_DESTROY`, what makes you believe that `WM_NCDESTROY` is the better choice?

Reply:
Because `WM_NCDESTROY` is also emitted if there are no child windows I believe.

Reply:
But `WM_NCDESTROY` being emitted doesn't seem sufficient as a reason to use it? That just seems like a coincidental fix.

Reply:
###### I'm (obviously) not @VZout, but...
I think you've misunderstood the docs.
When a window receives `WM_DESTROY`, three things are promised:

1. The window is being destroyed.
2. Any child windows still exist (but will soon be destroyed).
3. (implicitly) A `WM_NCDESTROY` message will be dispatched soon.

For `WM_NCDESTROY`, the following should hold true:

1. Any child windows are now destroyed
2. [MSDN](https://docs.microsoft.com/en-us/windows/win32/winmsg/wm-ncdestroy#remarks) "This message frees any memory internally allocated for the window."

This would suggest that the window can't receive any further messages after `WM_NCDESTROY` since Windows has presumably invalidated the window's handle, message queue, and other associated state (see [this SO answer](https://stackoverflow.com/a/41362844)).

Reply:
Your answer was more detailed anyway ♥